modules/web/client/src/main/scala/observe/ui/services/ConfigApiImpl.scala

```diff
@@ -20,18 +24,35 @@ import org.http4s.circe.*
 import org.http4s.client.Client
 import org.http4s.client.*
 import org.http4s.headers.Authorization
+import org.typelevel.log4cats.Logger
 
-case class ConfigApiImpl(client: Client[IO], baseUri: Uri, token: NonEmptyString)
+case class ConfigApiImpl(
+  client:        Client[IO],
+  baseUri:       Uri,
+  token:         NonEmptyString,
+  isBlockedView: View[Boolean],
+  latch:         Resource[IO, Unit],
+  onError:       Throwable => IO[Unit]
+)(using Logger[IO])
     extends ConfigApi[IO]:
-  private def request[T: Encoder](path: String, data: T): IO[Unit] = // TODO: Retries
-    client.expect[Unit](
-      Request(Method.POST, baseUri.addPath(path))
-        .withHeaders(Authorization(Credentials.Token(AuthScheme.Bearer, token.value)))
-        .withEntity(data.asJson)
+  private def request[T: Encoder](path: String, data: T): IO[Unit] =
+    latch.use(_ =>
+      isBlockedView.async.set(true) >>
+        client
+          .expect[Unit](
+            Request(Method.POST, baseUri.addPath(path))
+              .withHeaders(Authorization(Credentials.Token(AuthScheme.Bearer, token.value)))
+              .withEntity(data.asJson)
+          )
+          .onError(onError)
+          .onCancel(onError(new Exception("There was an error modifying conditions."))) >>
```

Comment:
I guess this error message is more general than just conditions?

Reply:
This is the configuration API, it is conditions + operator + observer. Will change to "configuration".